Sources/_InternalTestSupport/ProcessInfo+hostutils.swift

```diff
@@ -22,14 +22,16 @@ extension ProcessInfo {
                 return false
             }
         }
-        let lines = contentString.components(separatedBy: .newlines)
-        for line in lines {
-            if line.starts(with: "ID=") {
-                let id = line.replacingOccurrences(of: "ID=", with: "").trimmingCharacters(in: .whitespacesAndNewlines)
-                if id == "amzn" { // ID for Amazon Linux is "amzn"
+        do {
+            let nameRegex = try Regex("PRETTY_NAME=\"?Amazon Linux 2.*\"?")
```

Comment:
I suggest you just do an exact match without the regex -- as is, this will incorrectly match Amazon Linux 2023 (due to the greedy match), which is the next version of Amazon Linux that the Swift project will later switch to, and notably does _not_ have the posix_spawn issues that cause us to special case AL2 in the first place.

Reply:
I'd rather have this merged now and edge cases dealt with later in separate PRs. This PR is blocking toolchain builds and AL2 failures is one of the reasons we don't have published toolchains since 22nd of June.